diffrax/_solver/langevin_srk.py

```diff
@@ -0,0 +1,289 @@
+import abc
+from typing import Generic, TypeVar
+
+import equinox as eqx
+import jax.lax as lax
+import jax.numpy as jnp
+import jax.tree_util as jtu
+from jax import vmap
+from jax._src.tree_util import PyTreeDef
+from jaxtyping import Array, ArrayLike, PyTree
+
+from .._custom_types import (
+    AbstractBrownianIncrement,
+    BoolScalarLike,
+    DenseInfo,
+    RealScalarLike,
+)
+from .._local_interpolation import LocalLinearInterpolation
+from .._solution import RESULTS
+from .._term import _LangevinArgs, LangevinTerm, LangevinTuple, LangevinX
+from .base import AbstractItoSolver, AbstractStratonovichSolver
+
+
+_ErrorEstimate = TypeVar("_ErrorEstimate", None, LangevinTuple)
+
+
+class _AbstractCoeffs(eqx.Module):
+    @property
+    @abc.abstractmethod
+    def dtype(self):
+        raise NotImplementedError
+
+
+_Coeffs = TypeVar("_Coeffs", bound=_AbstractCoeffs)
+
+
+# TODO: I'm not sure if I can use the _Coeffs type here,
+# given that I do not use Generic[_Coeffs] in the class definition.
+# How should I work around this?
```

Comment:
What goes wrong with using `Generic` in the class definition?

Reply:
Hmm, I remember pyright complaining last time, but now I tried adding the `Generic` again and it works, so good I guess.